server/users/users.service.ts

```diff
@@ -69,21 +122,40 @@ class UserService {
 		return await this.getById(userId)
 	}
 
-	public async deleteUser(userId: number) {
-		const deleted = await Users.destroy({ where: { id: userId } });
+	public async updateUserSetting(userId: number, settingData: UpdateSettingCommand): Promise<UserModel> {
+		if (isEmptyObject(settingData)) {
+			throw new HttpException(400, "Incorrect input data");
+		}
 
-		if (!deleted) {
-			throw new HttpException(409, "User not found");
+		let settingUpdate = { ...{}, [settingData.setting]: (settingData.value == "true") };
+
+		const updated = await UserSettings.update(
+			settingUpdate,
+			{ where: { user_id: userId } }
+		);
+
+		if (!updated) {
+			throw new HttpException(409, "User settings not found");
 		}
+
+		return await this.getById(userId)
 	}
 
-	private mapToDto(model: Users): UserDto {
-		return {
-			id: model.id!,
-			email: model.email,
-			password: model.password,
-			isAdmin: model.is_admin,
+	public async deleteUser(userId: number, settingsId: number) {
+		const deletedSettings = await UserSettings.destroy({ where: { id: settingsId } });
```

Comment:
Can this be destroyed by `{ user_id: userId } ` like below entries, so we don't have to pass in `settingsId`?

Reply:
resolved in https://github.com/mbrizic/ratethegame/pull/28/commits/b158c0718600a8adeb6bacdcd5d09c7e8ab913e4